src/widget/wtrackproperty.cpp

```diff
@@ -1,18 +1,44 @@
+#include "widget/wtrackproperty.h"
+
+// Qt includes
+#include <QMenu>
+#include <QtCore/QItemSelectionModel>
+#include <QtWidgets/QInputDialog>
+#include <QtWidgets/QCheckBox>
+#include <QtWidgets/QWidgetAction>
 
-#include <QDebug>
-#include <QUrl>
+// std includes
+#include <utility>
 
+// Project includes
 #include "control/controlobject.h"
-#include "widget/wtrackproperty.h"
+#include <library/crate/cratefeaturehelper.h>
+#include "library/dao/playlistdao.h"
+#include "library/trackcollection.h"
+#include "library/trackcollectionmanager.h"
+#include "library/trackmodel.h"
+#include "track/track.h"
 #include "util/dnd.h"
+#include "util/desktophelper.h"
+#include <iostream>
 
 WTrackProperty::WTrackProperty(const char* group,
                                UserSettingsPointer pConfig,
-                               QWidget* pParent)
+                               QWidget* pParent,
+                               TrackCollectionManager* pTrackCollectionManager)
         : WLabel(pParent),
           m_pGroup(group),
           m_pConfig(pConfig) {
     setAcceptDrops(true);
+
+    // Setup context menu
+    WTrackMenu::Filters flags = WTrackMenu::Filter::Playlist |
+                                WTrackMenu::Filter::Crate;
```

Comment:
add Metadata, Reset, Adjust BPM, Set Color, and Properties here

Reply:
Adding them will need some work in the WTrackMenu since some actions are still dependent on a track model. Metadata, reset, adjust bpm, and properties are inherent properties of the track, but set color changes the color for a row in a WTrackTableView, so it can't be added here.

Reply:
Okay so let's work on removing WTrackMenu's dependency on TrackModel.

> set color changes the color for a row in a WTrackTableView, so it can't be added here.

I don't see a dependency on TrackModel in WTrackMenu::slotColorPicked.

Reply:
It sure doesn't depend on TrackModel. I saw the colours reflecting only in the WTrackTableView so I assumed it should be available only to a WTrackTableView, but the colour information can certainly be used by other widgets independent of the track model.
I will add this to WTrackProperty.